src/wrapper.jl

```diff
@@ -319,24 +319,11 @@ function _LGBM_DatasetSetField(ds::Dataset, field_name::String,
     return nothing
 end
 
-function LGBM_DatasetSetField(ds::Dataset, field_name::String, field_data::Vector{Float32})
-    if field_name == "label" || field_name == "weight"
-        _LGBM_DatasetSetField(ds, field_name, field_data)
-    else
-        _LGBM_DatasetSetField(ds, field_name, convert(Vector{Int32}, field_data))
-    end
-    return nothing
-end
-
-function LGBM_DatasetSetField(ds::Dataset, field_name::String, field_data::Vector{Int32})
-    if field_name == "group"
-        _LGBM_DatasetSetField(ds, field_name, field_data)
-    else
-        _LGBM_DatasetSetField(ds, field_name, convert(Vector{Float32}, field_data))
-    end
-    return nothing
-end
-
+# the below type conversions are required by C API for the different fieldnames
+# ideally we should enforce the correct types through signature and multiple dispatch accordingly
+# but this would restrict the inputs from MLJInterface which don't necessarily have
+# exact matching data types for these fields hence leaving the conversions here before passing to the
+# wrapper function
 function LGBM_DatasetSetField(ds::Dataset, field_name::String, field_data::Vector{T}) where T<:Real
```

Comment:
ideally here we should have something like this (with or without the error handling and may be even with structs for field_name):

```
function LGBM_DatasetSetField(ds::Dataset, field_name::String, field_data::Vector{Float32})
    if field_name == "label" || field_name == "weight"
        _LGBM_DatasetSetField(ds, field_name, field_data)
    else
        error("Invalid field name: $field_name for Float32 data")
    end
    return nothing
end

function LGBM_DatasetSetField(ds::Dataset, field_name::String, field_data::Vector{Float64})
    if field_name == "init_score"
        _LGBM_DatasetSetField(ds, field_name, field_data)
    else
        error("Invalid field name: $field_name for Float64 data")
    end
    return nothing
end

function LGBM_DatasetSetField(ds::Dataset, field_name::String, field_data::Vector{Int32})
    if field_name == "group"
        _LGBM_DatasetSetField(ds, field_name, field_data)
    else
        error("Invalid field name: $field_name for Int32 data")
    end
    return nothing
end

```

but this seems too restrictive especially for labels coming from interfaces calling lgbm which don't have such type restrictions
```